Tutorials/whatiswot/11-Architecture_Patterns/11-Architecture-Patterns.vtt

```diff
@@ -0,0 +1,107 @@
+WEBVTT
+
+00:00:00.000 --> 00:00:12.662 
+In this video, we will explain architecture patterns, and how devices and things interact with controllers, agents, and servers.
+
+00:00:12.662 --> 00:00:23.356 
+Previously we have explained the server and client models and described how this architecture works.
+
+00:00:23.356 --> 00:00:27.481 
+Now we will show how they are used in the WoT architectures.
+
+00:00:27.481 --> 00:00:34.714 
+We will introduce 6 architecture patterns in this video.
+
+00:00:34.714 --> 00:00:42.120 
+So, first, let’s start with the device controllers. A common deployment pattern is a local device
+
+00:00:42.120 --> 00:00:49.526 
+controlled by a user-operated remote controller. Let’s remember the smart house example from the previous video.
+
+00:00:49.526 --> 00:00:55.426 
+A remote controller can interact with the electronic devices through the local home network.
+
+00:00:55.426 --> 00:01:02.576 
+In this pattern, at least one electronic device has a server role that can accept a request and respond.
+
+00:01:02.576 --> 00:01:08.008 
+Another device remote controller has a client role that can send a message with
+
+00:01:08.008 --> 00:01:13.439 
+a request like to read a sensor value or to turn on the device.
+
+00:01:13.439 --> 00:01:18.620 
+Let’s move to the second architecture pattern, Thing-to-Thing. In this scenario we
+
+00:01:18.620 --> 00:01:23.801 
+have an electronic device, let's say a heater and a temperature sensor.
+
+00:01:23.801 --> 00:01:31.374 
+When a sensor detects a change in room condition such as temperature drops below the threshold,
+
+00:01:31.374 --> 00:01:38.947 
+it issues a control message “turn on” to the electronic device to power on the heater.
+
+00:01:38.948 --> 00:01:43.973 
+Third architecture that we will introduce is Gateways.
+
+00:01:43.973 --> 00:01:51.404 
+Smart home gateways pattern is achieved by placing a gateway between a home network and the Internet to
+
+00:01:51.404 --> 00:01:58.835 
+manage electronic devices inside the house. It can also receive commands from a remote controller over the Internet.
```

Comment:
```suggestion
manage devices inside the house. It can also receive commands from a remote controller over the Internet.
```

Generally we do not need to say electronic.